packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -0,0 +1,434 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { assert } from "@fluidframework/common-utils";
+import { Value, Anchor } from "../../tree";
+import {
+	IEditableForest, TreeNavigationResult, mapCursorField, ITreeSubscriptionCursor, ITreeSubscriptionCursorState,
+} from "../../forest";
+import { brand } from "../../util";
+import { TreeSchema, FieldSchema, rootFieldKey, LocalFieldKey, TreeSchemaIdentifier } from "../../schema-stored";
+import { FieldKind, Multiplicity } from "../modular-schema";
+import {
+    AdaptingProxyHandler,
+    adaptWithProxy,
+    getFieldKind, getFieldSchema, getPrimaryField, isPrimitive, isPrimitiveValue, PrimitiveValue,
+} from "./utilities";
+
+// Symbol for extracting target from editable-tree proxies.
+// Useful for debugging and testing, but not part of the public API.
+export const proxySymbol: unique symbol = Symbol("editable-tree proxy");
+
+/**
+ * A symbol to for the type of a node in contexts where string keys are already in use for fields.
+ */
+export const type: unique symbol = Symbol("editable-tree type");
+
+ /**
+ * A symbol to for the value of a node in contexts where string keys are already in use for fields.
+ */
+export const value: unique symbol = Symbol("editable-tree value");
+
+/**
+ * A symbol to for the type name of a child node in contexts where string keys are already in use for fields.
+ */
+ export const getTypeName: unique symbol = Symbol("editable-tree typeName");
```

Comment:
I'd prefer to avoid having this.

I want to encourage everyone at this abstraction level to use the schema object instance instead of the type name. This makes handling stored vs view schema easier (you never can accidently use the type name to look up the schema and get the wrong one). Its also more forwar
```suggestion
/**
 * A symbol to for the type name of a child node in contexts where string keys are already in use for fields.
 *
 * TODO:
 * Add name to TreeSchema retuned via `type` and remove this.
 * We want to encourage using the TreeSchema instances and not the string names to improve robustness when multiple schema have the same name (for example a stored schema and a view schema which mismatch, or when if later add support for contextual schema).
 * Also using the schema instance directly should be faster for a lot of use cases (reference comparison should be faster than string comparison, and access to data from schema will avoid a map lookup)
 */
 export const getTypeName: unique symbol = Symbol("editable-tree typeName");
```

Reply:
Thats probably going to need some line wrapping to not fail the linter.

Reply:
As for the Inspector usecase, we actually don't need anything other than a type name for UI. A field schema might be needed later, but we need build-in options to narrow down this complexity for simple usecases. I'll convert into a function and add some options.

Reply:
Another issue is to get types for primitive fields. So, anyway it should be a function with at least a key to get types of the children.

Reply:
I intended that usecase (getting types of children) to be covered by a couple different approaches:

* If you want the schema for the field (what types are allowed there), you can get the schema for the node and look up the type of the field. This already works (though due to annoyances with local vs global field key aliasing, it's not a great API yet).
* If you want the type of the child node, we will provide an API that lets you get the child node without unwrapping (ex, maybe via a "getWrappedChild" symbol or something). There is a TODO to add such an API on EditableTree (copied below).
* We could also provide an API that lets you access the field without unwrapping. For now this would just give you an array, but once we have field proxies, there could be an API on it to get the field schema (as well as an API to get children with or without unwrapping). I think this also falls under the below TODO.

```
TODO: provide non-schema impacted APIs for getting fields and nodes without unwrapping
 * (useful for generic code, and when references to these actual fields and nodes are required,
 * for example creating anchors and editing).
```